providers/core.tcpsocket.js

```diff
@@ -75,15 +75,10 @@ Socket_chrome.prototype.connect = function(hostname, port, cb) {
   chrome.sockets.tcp.create({}, function(createInfo) {
     this.id = createInfo.socketId;
     try {
-      chrome.sockets.tcp.connect(this.id, hostname, port, function (result) {
-        if (result < 0) {
-          var errorString = Socket_chrome.errorStringOfCode(result);
-          var freedomErrorCode = Socket_chrome.freedomErrorCode(errorString);
-          cb(undefined, {
-            'errcode': freedomErrorCode,
-            'message': 'Chrome Connection Failed: ' + errorString +
-                ' ' + chrome.runtime.lastError.message
-          });
+      chrome.sockets.tcp.connect(this.id, hostname, port, function(result) {
+        var error = Socket_chrome.chromeErrorHandler(result);
```

Comment:
I think @soycode's solution is fine.  Checking `code < 0 || chrome.runtime.lastError` also seems reasonable.